nicsdru_nidirect_theme.theme

```diff
@@ -328,16 +328,36 @@ function nicsdru_nidirect_theme_preprocess_page_title(&$variables) {
   // By default page titles are visible.
   $variables['title_visible'] = TRUE;
 
-  // Preprocess node titles.
-  if ($node = \Drupal::routeMatch()->getParameter('node')) {
-
-    // Landing pages have a boolean field to enable title visibility.
-    if ($node->getType() == 'landing_page') {
-      // Is the title visible?
-      if ($node->field_enable_title->get(0)->value === '0') {
-        $variables['title_visible'] = FALSE;
+  // Preprocess titles on certain routes.
+  $route = \Drupal::routeMatch();
+
+  switch ($route->getRouteName()) {
+
+    // Preprocess node titles.
+    case "entity.node.canonical":
+      $node = $route->getParameter('node');
+      // Landing pages have a boolean field to enable title visibility.
+      if ($node->getType() === 'landing_page') {
+        // Is the title visible?
+        if ($node->field_enable_title->get(0)->value === '0') {
```

Comment:
It might be a good idea to defensively check whether 'field_enable_title' exists and has an item [0] so that there is no possibility of this line causing an error?

Reply:
Fixed hopefully - although I had to remove line 266 which was setting `$variables['show_title'] = FALSE;` for landing pages.  Not sure why though - comment above line 266 says: 
```
// Omit the title on these content types; causes trouble with search
// indexing.
```

Reply:
Looks good